### ISSUE TITLE
Error convert CSV exported from database to Parquet (empty/null values)

### ISSUE THREAD

**Issue by Thiago-Dantas** [awslabs/aws-data-wrangler#333]
We are trying to convert CSV files exported from relational databases with AWS DMS
We have encountered a number of different errors trying to convert many different types of files
We also have the Glue Catalog setup directly from source database

INT columns with null values break, pandas try to set them as pd.NaN(float) and eventually everything breaks due to failed conversion. On our tests replacing `np.NaN` to `''` fixes this 
BOOLEAN with null values break, this one breaks in `_cast_pandas_column df[col] = df[col].astype(desired_type)`  with error `TypeError('Need to pass bool-like values',)`. This one tries to cast to type `'boolean'` but if I change `desired_type` during debugging to `'bool'`(python type) it works

This is our own "fix it" function, but we are still stuck dealing with broken Date types and it raises the question if this should be necessary at all

```
def prepare_dataframe(df, dtypes):
    for c in df.columns:
        if(dtypes[c] in ['bigint','int']):
            df[c] = df[c].replace(np.NaN , '')
        if(dtypes[c] in ['float','double']):
            df[c] = df[c].replace('',np.NaN)
            df[c] = df[c].astype(float)
        if(dtypes[c] in ['boolean']):
            df[c] = df[c].astype(bool)
    return df
```

**Comment by igorborgest** [comment]
Hi @Thiago-Dantas,

What are you trying to do when errors pop up? Trying to call `wr.s3.to_parquet()`?
If yes, what arguments are you passing?

**Comment by Thiago-Dantas** [comment]
Yes, errors pop up on `s3.to_parquet` when passing df, path, use_threads=True, dataset=True, database,table(with a Glue Catalog previously built based on a relational database) and compression
```
    response = wr.s3.to_parquet(   df=final_df, 
                            path=parquet_path , 
                            use_threads=True, 
                            dataset=True,
                            database= database, 
                            table= table,
                            compression='snappy')
```

We got it working currently by settings `dtype` based on Glue Catalog same as awswrangler, removing the dataset,database and table parameters but "fixing" the dataframe prior to calling `to_parquet` with our `prepare_dataframe` function
```
    response = wr.s3.to_parquet(   df=final_df, 
                            path=parquet_path , 
                            use_threads=True, 
                            dataset=False,
                            compression='snappy',
                            dtype=dtypes)
```

**Comment by igorborgest** [comment]
What is the error message for the first case (integer columns with np.NaN) ?

**Comment by igorborgest** [comment]
Closing by inactivity.

**Comment by Thiago-Dantas** [comment]
Just for reference, I was unable to reproduce our initial issues with INT and BIGINT data types

using `.astype(bool)` can be the wrong choice since in python `bool("") == False` and thats not necessarily true

We got it fully working by tweaking our read_csv function with pandas_kwargs
```
    pandas_kwargs = {
        'sep':'|',
        'keep_default_na': False,
        'true_values': ['true','True','1'],
        'false_values': ['false','False','0'],
        'na_values': ''
    }
```
And then "fixing" the dataframe with 
```
def prepare_dataframe(df, types):
    for c in df.columns:
        if(types.get(c) in ['bigint','int']):
            df[c] = df[c].replace('', pd.NA)
        if(types.get(c) in ['float','double']):
            df[c] = df[c].replace('', np.NaN)
        if(types.get(c) in ['string']):
            df[c] = df[c].replace(np.NaN, '').astype(str)
    return df
```
With our tweak on read_csv, empty string values received np.NaN value which converts to string "nan", thus needs fixing
`.astype(str)` fixes the error `StringArray requires a sequence of strings or pandas.NA` we were receiving occasionally

**Comment by igorborgest** [comment]
Thanks for share @Thiago-Dantas .
In addition, Pandas just [released the version `1.1.0` with the support for cast anything to String](https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.1.0.html#all-dtypes-can-now-be-converted-to-stringdtype) (Already in Wrangler 1.7.0) it should help you to overcome the last situation that you said.

**Comment by igorborgest** [comment]
Hi @Thiago-Dantas !

It is totally off-topic, but we are stating a [_"Who uses AWS Data Wrangler?"_](https://github.com/awslabs/aws-data-wrangler/tree/dev#who-uses-aws-data-wrangler) section. So feel free to add yourself if you want 😄 .
